### DISCUSSION TITLE
Replace noun_chunks with root

### DISCUSSION THREAD

**Discussion by webecker25272** [explosion/spaCy#7260] [Help: Coding & Implementations]
Have a bunch of documents that have text like this:
`The very nice product is convenient to use`

The noun chunker is combining "the very nice product" into a single chunk, which is PERFECT.

What I want to to do is eliminate the non-root elements of the chunk.  So the desired output should be:
`Product is convenient to use` 

How can I replace the whole span of elements 1-4?  Is this something I should do in a custom pipeline component?  If so, I couldn't figure out how to do that.  Does this possibly involve retokenization?

Thanks!!

**Comment by adrianeboyd** [comment]
One of spaCy's core principles is that it is non-destructive with respect to the input text, so pipeline components are only able to modify annotation, not the text itself. Retokenization can only change the token boundaries, not the text.

You can create a new string text or a new `Doc` with a modified text/tokens outside of the current pipeline, but you won't be able to modify the text itself within a single pipeline.

**Reply by webecker25272** [reply]
That's super helpful thank you @adrianeboyd !!!!
